### PR TITLE
conf: add userns_exec_full()

### LINKED ISSUE #1800: userns_exec_1: use full idmap for copying container storage
We need all ids mapped since we don't perform a minimal operation. Instead, we need full control over all sub{g,u}ids so that e.g. `rsync` can happily `chown` away.

### PR DESCRIPTION
**Pull request by brauner**
Closes #1800.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
jenkins: test this please

File: src/lxc/utils.c [hunk 1]
```diff
@@ -2330,9 +2330,10 @@ int run_command(char *buf, size_t buf_size, int (*child_fn)(void *), void *args)
 	/* close the write-end of the pipe */
 	close(pipefd[1]);
 
-	bytes = read(pipefd[0], buf, (buf_size > 0) ? (buf_size - 1) : 0);
-	if (bytes > 0)
+	if (buf && buf_size > 0) {
+		bytes = read(pipefd[0], buf, buf_size - 1);
 		buf[bytes - 1] = '\0';
```
Comment (hallyn) [inline src/lxc/utils.c]: If read returns -1, you'll do buf[-2] = '\0';

Comment (brauner) [reply src/lxc/utils.c]: Rigt, thanks for spitting this!

Comment (hallyn) [reply src/lxc/utils.c]: <spit>

Comment (hallyn) [reply src/lxc/utils.c]: Iiiinteresting.  Let's try that again.  What happens when I type:

"<spit>"

Comment (hallyn) [reply src/lxc/utils.c]: <blink>Look, I'm annoying</blink>
I really didn't think comments did this sort of thing.  I'm disappointed.

Comment (brauner) [reply src/lxc/utils.c]: s/spitting/spotting/
Damn autocorrect on the phone.

**Pull request merged**